Update release instructions (#1424)
mongodb/mongo-cxx-driver: etc/releasing.md

git commit -m 'Update CHANGELOG for X.Y.Z'
```

> ##### Pre-Release Changes PR

Reviewer: Suggest removing this section header entirely. No reason to separate it from the "... for a Non-Patch Release" section body.

Example (using Jira syntax formatting):

```
* The [Compatibility|https://www.mongodb.com/docs/languages/cpp/cpp-driver/current/compatibility/] page must be updated:

>   - mongocxx-X.Y.Z changes "libmongoc Compatibility" and requires mongoc-A.B.C or later.
>   - mongocxx-X.Y.Z does has the same "MongoDB Compatibility", "Language Compatibility", or "Compiler Compatibility" as version mongocxx-X.Y.(Z-1)

Reviewer: ```suggestion
  - mongocxx-X.Y.Z may change "libmongoc Compatibility" by requiring mongoc-A.B.C or later.
  - mongocxx-X.Y.Z should have the same "MongoDB Compatibility", "Language Compatibility", and "Compiler Compatibility" as version mongocxx-X.Y.(Z-1)
```

Wording tweak suggestions.
